crates/sui-core/src/authority.rs

```diff
@@ -1018,6 +1018,10 @@ impl AuthorityState {
         transaction: TransactionData,
         transaction_digest: TransactionDigest,
     ) -> Result<SuiTransactionEffects, anyhow::Error> {
+        if !self.is_fullnode() && !cfg!(test) {
```

Comment:
@lxfind I could not find an easy way to make a fullnode for tests, hence this monstrosity. Any suggestions?

Reply:
Not sure what does "!cfg!(test)" mean here? We need this API in prod, right?

Reply:
You had suggested this `!self.is_fullode()` check, that is to error for validators handling this request. That is not easily achievable since all of our integration tests are using validators.

Reply:
Do you mean our integration tests for dev queries? In that case, we should fix the integration tests to use fullnodes then?

Reply:
> In that case, we should fix the integration tests to use fullnodes then?

That's the part where I got stuck :) But I can take a closer look again and see if I can get more specific questions

Reply:
Well that was a bit more than I expected, and it's a tad messy. But I did find a bug in some existing tests :) 

This `!cfg!(test)` check is now gone, with authority tests being a tad more complicated